src/components/UI/Display/Apps/AppList/AppGrid.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { memo, useLayoutEffect, useRef, useState } from 'react';
+import { WindowScroller } from 'react-virtualized';
+import {
+  areEqual,
+  GridChildComponentProps,
+  VariableSizeGrid,
+} from 'react-window';
+import styled from 'styled-components';
+import { IAppProps } from 'UI/Display/Apps/AppList/App';
+
+interface IAppGrid {
+  items: IAppProps[];
+  itemMinWidth: number;
+  itemMinHeight: number;
+  gridGap: number;
+  render: (item: IAppProps, i: number) => React.ReactNode;
+}
+
+interface IScrollPosition {
+  scrollTop: number;
+}
+
+interface IDataProps {
+  items: IAppProps[];
+  columnCount: number;
+  gridGap: number;
+  render: (item: IAppProps, i: number) => React.ReactNode;
+}
+
+const StyledContainer = styled.div<{ gridGap: number }>`
+  margin: 0 -${(props) => props.gridGap / 2}px;
+`;
+
+const StyledGrid = styled(VariableSizeGrid)<{ minHeight: number }>`
+  height: 100% !important;
+
+  > div {
+    height: ${(props) => props.minHeight}px !important;
+  }
+`;
+
+const getIndex = (
+  rowIndex: number,
+  columnIndex: number,
+  columnCount: number
+) => {
+  return rowIndex * columnCount + columnIndex;
+};
+
+const ItemRenderer: React.FC<GridChildComponentProps> = memo(
+  ({ columnIndex, rowIndex, data, style }) => {
+    const { items, columnCount, render, gridGap }: IDataProps = data;
+    const index = getIndex(rowIndex, columnIndex, columnCount);
+    const item = items[index];
+
+    const patchedStyle = Object.assign({}, style, {
+      padding: `0 ${gridGap / 2}px`,
+    });
+
+    return item ? <div style={patchedStyle}>{render(item, index)}</div> : null;
+  },
+  areEqual
+);
+
+const AppGrid: React.FC<IAppGrid> = ({
+  items,
+  itemMinWidth,
+  itemMinHeight,
+  gridGap,
+  render,
+}) => {
+  const [width, setWidth] = useState<number>(0);
+
+  const gridRef = useRef<VariableSizeGrid>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const handleScroll = ({ scrollTop }: IScrollPosition) => {
+    if (gridRef.current) gridRef.current.scrollTo({ scrollTop });
+  };
+
+  useLayoutEffect(() => {
+    const handleResize = () => {
```

Comment:
This is not entirely necessary right now, since performance is great, but for future improvement:
This function could be debounced, so we won't be calling `getBoundingClientRect()` so often, since that triggers reflows